bundler/lib/bundler/definition.rb

```diff
@@ -281,7 +281,7 @@ def groups
     end
 
     def lock(file, preserve_unknown_sections = false)
-      return if Definition.no_lock
+      return if Definition.no_lock || file.nil?
```

Comment:
I think this removes the need for `file &&` on line 301, no? (Unless `file` can also be `false` for some reason?)

Reply:
Indeed, thanks!

Reply:
This comment still makes sense, we can now stop using `file_exists?(file)` a few lines below and do `File.exist?(file)` instead since the `nil` check is no longer needed. After that,  the `file_exists?` method has a single caller, so can probably be removed and change `lockfile_exists` (the only caller) to instead do `lockfile && File.exist?(lockfile)`.